Keyboards/KeyboardsBase/KeyboardKeys.swift

```diff
@@ -291,25 +283,46 @@ class KeyboardKey: UIButton {
     } else if DeviceType.isPad {
       adjustPadKeyWidth()
     }
+  }
 
+  /// Adjusts the style of the button based on different states
+  func adjustButtonStyle() {
     guard let isSpecial = layer.value(forKey: "isSpecial") as? Bool else { return }
 
-    if key == "shift" {
-      // Switch the shift key icon given its state.
+    switch key {
+    case SpecialKeys.indent:
+      backgroundColor = specialKeyColor
+
+    case SpecialKeys.capsLock:
+      switch capsLockButtonState {
+
+      case .normal:
+        backgroundColor = specialKeyColor
+        styleIconBtn(btn: self, color: UIColor.label, iconName: "capslock")
+
+      case .locked:
+        backgroundColor = keyPressedColor
+        styleIconBtn(btn: self, color: UIColor.label, iconName: "capslock.fill")
+      }
+
+    case "shift":
       if shiftButtonState == .shift {
         backgroundColor = keyPressedColor
         styleIconBtn(btn: self, color: UIColor.label, iconName: "shift.fill")
-      } else if shiftButtonState == .caps {
-        backgroundColor = keyPressedColor
-        styleIconBtn(btn: self, color: UIColor.label, iconName: "capslock.fill")
       } else {
         backgroundColor = specialKeyColor
       }
-    } else if key == "return" && [.translate, .conjugate, .plural].contains(commandState) {
-      // Color the return key depending on if it's being used as enter for commands.
-      backgroundColor = commandKeyColor
-    } else if isSpecial {
-      backgroundColor = specialKeyColor
+
+    case "return":
+      if [.translate, .conjugate, .plural].contains(commandState) {
+        // Color the return key depending on if it's being used as enter for commands.
+        backgroundColor = commandKeyColor
+      }
```

Comment:
We need the else back here for one bug I'm about to mention such that we get `backgroundColor = specialKeyColor`.

Reply:
There's a default case which should handle the `else`but I'll do a review of my code to see if it's not missing some cases.